homeassistant/components/digital_ocean/__init__.py

```diff
@@ -81,6 +89,31 @@ def get_droplet_id(self, droplet_name):
 
         return droplet_id
 
+    @Throttle(MIN_TIME_BETWEEN_DOMAIN_UPDATES)
+    def update_domain_record(
+        self, domain_name, record_name, record_value, record_type="A"
+    ):
+        """Update the appointed DNS record with the desired value."""
+        domain = digitalocean.Domain(token=self._access_token, name=domain_name)
+        try:
+            records = domain.get_records()
+        except digitalocean.baseapi.NotFoundError as exc:
+            raise DomainRecordsNotFound(
+                f"Could not find records in domain {domain_name}"
+            ) from exc
+
+        for record in records:
+            if record.name == record_name and record.type == record_type:
+                if record.data == record_value:
+                    return False
+                record.data = record_value
+                record.save()
+                return True
+        raise DomainRecordsNotFound(
```

Comment:
Is there a reason why this is wrapped in a custom exception?

Reply:
(see above)

Reply:
Marking this one as resolved, since the discussion is being carried above.